tests/CawFunctionTests.cpp

```diff
@@ -0,0 +1,261 @@
+#include "../src/caw/functions/caw_functions.h"
+#include "../src/key_value_store/core/KeyValueStore.h"
+
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+KeyValueStore store;
+KeyValueStoreInterface &kvstore = store;
```

Comment:
For repeated-use components in a set of gTests, use test fixtures with `TEST_F`, as shown here: https://github.com/google/googletest/blob/master/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests-same-data-multiple-tests.

Reply:
So I looked into this before writing my tests. The way I see it is that the data inside the class persists, but data generated inside the test is erased (because they destroy and remake the class after every test function). Currently some tests build on others using some information put into the kvstore by the other test. 

For example my read child caw test only works because I have previously posted multiple child caws in the post caw tests. Would you recommend I keep the tests the way they are or input the data from scratch in each test function?